karapace/kafka/producer.py

```diff
@@ -59,3 +62,41 @@ def partitions_for(self, topic: str) -> dict[int, PartitionMetadata]:
             return self.list_topics(topic).topics[topic].partitions
         except KafkaException as exc:
             raise_from_kafkaexception(exc)
+
+
+class AIOKafkaProducer:
+    """An async wrapper around `KafkaProducer` built on top of confluent-kafka.
+
+    Starting `AIOKafkaProducer` instantiates a `KafkaProducer` and starts a poll-thread.
+
+    The poll-thread continuously polls the underlying producer so buffered messages
+    are sent and asyncio futures returned by the `send` method can be awaited.
+    """
+
+    def __init__(
+        self,
+        bootstrap_servers: Iterable[str] | str,
+        loop: asyncio.AbstractEventLoop | None = None,
+        **params: Unpack[KafkaClientParams],
+    ) -> None:
+        self.loop = loop or asyncio.get_running_loop()
+        self.producer = KafkaProducer(bootstrap_servers, **params)
+
+        self.stopped = Event()
+        self.poll_thread = Thread(target=self.poll_loop)
+        self.poll_thread.start()
+
+    async def stop(self) -> None:
+        self.stopped.set()
+        self.poll_thread.join()
+
+    def poll_loop(self) -> None:
```

Comment:
Yes, but also for any messages to be sent from the local buffer `poll` or `flush` (which is just `poll` in a loop until the buffer is empty) needs to be called. To achieve `asyncio` compatible behaviour then, `poll` needs to be called constantly. Otherwise the client would have to take care of this, which doesn't fit into the whole async setup.